src/interfaces.ts

```diff
@@ -104,32 +104,34 @@ type unstable_WacAllow = {
   user: unstable_AccessModes;
   public: unstable_AccessModes;
 };
+
 /**
  * [[LitDataset]]s fetched by lit-pod include this metadata describing its relation to a Pod Resource.
  */
-export type ResourceInfo = {
-  resourceInfo: {
-    fetchedFrom: UrlString;
-    /**
-     * The URL reported by the server as possibly containing an ACL file. Note that this file might
-     * not necessarily exist, in which case the ACL of the nearest Container with an ACL applies.
-     *
-     * @ignore We anticipate the Solid spec to change how the ACL gets accessed, which would result
-     *         in this API changing as well.
-     */
-    unstable_aclUrl?: UrlString;
-    /**
-     * Access permissions for the current user and the general public for this resource.
-     *
-     * @ignore There is no consensus yet about how this functionality will be incorporated in the
-     *         final spec, so the final implementation might influence this API in the future.
-     * @see https://github.com/solid/solid-spec/blob/cb1373a369398d561b909009bd0e5a8c3fec953b/api-rest.md#wac-allow-headers
-     * @see https://github.com/solid/specification/issues/171
-     */
-    unstable_permissions?: unstable_WacAllow;
-  };
+export type ResourceWithInfo = {
+  resourceInfo: ResourceInfo;
 };
 
+export type ResourceInfo = {
```

Comment:
For everything that we `export`, we need to consider whether we want to expose it to developers. If we do, we should also export it from `index.ts`; if not, we need to explicitly exclude it from the API docs using `@hidden` or `@ignore`.

I'm not actually sure whether we want to hide it from developers; it could be useful when they're e.g. mocking Resources in tests? Then again, `ResourceWithInfo` already _is_ public.

```suggestion
/** @hidden Developers shouldn't create this object manually. */
export type ResourceInfo = {
```

By the way, in case you didn't know: as an alternative to splitting this out, the return type of `fetchResourceInfo` could also have been `Promise<ResourceWithInfo['resourceInfo']>`. Whatever you prefer :)

Reply:
I wasn't familiar with the `ResourceWithInfo['resourceInfo']` notation. I saw you using it a bunch of times, but not having used it myself I didn't think about it. It seems to be the simplest in this case, instead of exporting two very similar types